### ISSUE TITLE
Need help implementing list docker images outside of example

### ISSUE THREAD

**Issue by Darrilla** [vv9k/docker-api-rs#31]
I am trying to implement the list docker images example in my own project, but something is wrong. I am missing something fundamental and would appreciate a review of this code:
 
``` 
  async fn ListDockerImages() -> Result<(), Box<dyn std::error::Error>> {
  let docker = Docker::unix("/var/run/docker.sock");
  use docker_api::opts::ImageListOpts;
  let opts = ImageListOpts::builder().all(true).build();
  match docker.images().list(&opts).await {
    Ok(images) => {
      images.into_iter().for_each(|image| {
        println!(
          "---------------------------------\nCreated: {}\nId: {}\nRepo tags: {}\nLabels:\n{}",
        image.created,
        image.id,
        image.repo_tags.join(","),
        image
        .labels
        .into_iter()
        .map(|(k, v)| format!(" - {}={}", k, v))
        .collect::<Vec<_>>()
        .join("\n"),
        );
      });
    }
    Err(e) => eprintln!("Problem listing docker images: {}", e),
  }

  Ok(())
}

#[tokio::main]
async fn main() {
    ListDockerImages().await;
}
```

When I run this program, I get ```Problem listing docker images: invalid value: integer `-1`, expected usize at line 1 column 17```, not the actual listing I get when running the `cargo run --example image -- list`.

What am I doing wrong that the example code is doing correctly?  Thanks in advance.

**Comment by Darrilla** [comment]
Update: I was able to use the unreleased master branch code, and it works properly. I added this to my manifest:

[patch.crates-io]
docker-api = { path = "../docker-api-rs" }

**Comment by vv9k** [comment]
As you noticed the issue has been resolved in the master branch. I just released version `0.12.0` that should fix this issue.

**Comment by Darrilla** [comment]
> Thank you for the report! As you've noticed the issue has been resolved in the master branch. I just released version `0.12.0` that should fix this issue.

@vv9k You might want to update the README to specify using 0.12.0 in the Cargo.toml dependencies.

**Comment by vv9k** [comment]
Yes, thanks for spotting this! Fixed in b98596971ddecabbb2a15db016cc46d4caac2e3c (0.12.1)
